### ISSUE TITLE
Data loss due to unreliable sync and lots of missing version change history

### ISSUE THREAD

**Issue by BloodyIron** [nextcloud/notes#1207]
I have now lost multiple notes due to sync between either a desktop or android tablet not quite working right, and the version history being barely any entries. I just wrote like 10 questions for a job interview I'm about to do that should have been sync'd back to the server multiple times, but by the time I got back to my desktop, I lost half of these questions because the desktop version got synced overtop of the tablet work, and NONE of the "version history" versions for the file itself (when going into the Files function) has ANY of the data lost.

And this ISN'T the first time this has happened!

A sync ecosystem like this cannot have this much data loss, and frankly if it keeps up being this egregiously unreliable, I'm going to just stop using Notes through nextCloud completely.

Data loss for things like this cannot be acceptable, especially overwritting back onto the tablet like I've described. I could understand losing maybe a few characters or something, but ugh.

Anyways, this is clearly a bug composing of multiple aspects, including the incomplete Version Control aspect.

**Comment by stefan-niedermann** [comment]
Well, first of all: Pity, that should of course not happen (and given there is a user base od approximately 20.000 users this *does* not happen on a regular base).

Second: Data loss is always bad, no matter whether it was cause by defect hardware, a bug in the OS, web server, NC, an App, a sync client or an ecological catastrophe. Not having a reliabke backup strategy is bad, too. Let me note that this is part of *your* responsibility (to be exact, the one of the administrator, assuming this *is* you).

However I agree that such a behavior is clearly a bug and one thar should get fixed. So how did you imagine that wthe issue can be rproduced when you ignored the issue template, do not provide *any* information about your environment, used version, configurations (e. g. used editor) and so on? Why did you not copy the synchronization error message and post it here? I think there's a big fat <kbd>Copy</kbd> button next to it. If you want help and enhance the situation, you need to be more verbose, at its current state this issue is not more than a rant. Legit, but won't change anything.

**Comment by BloodyIron** [comment]
As for reproducing the issue, I was actually later on (a few hours?) able to get the Android device to "recover" the version that got over-written. But I do not know why it was so much later that the Android version of Notes detected this. The Ansdroid device at the time (and generally almost all the time) is on the same LAN as the nextCloud system, my WIFI is extremely reliable (UAP-AC-PRO), so connectivity was good. It didn't really make sense, but there were some connectivity issues that seemed to be within the app itself (as I was refreshing the app as I was adding to the list of questions).

So from my UX perspective, I would have _expected_ (so to say) nextCloud to have had more versions in the recent history for me to restore from (with the Files app, outside of Notes). Considering I had "told" the Notes app to sync contents in the realm of 8 times over the span of writing the contents.

So as to how to reproduce it, that's a tricky one, as the app seemed to maybe be acting in a way that was unexpected (the connectivity issues) as I really did not have those issues previously. And considering that the app effectively told me sync was successful each time (spinning circle finish spinning and dismisses without error) my "expectation" was the content was on the nextCloud server (when it wasn't). Which makes reproduction hard as nails IMO.

As for my Enivronment:
* nextCloud: v27.1.5 running on LAMP stack in a VM
* Notes App (server-side): v4.9.2
* Editor, I think I was using the rich text editor on Android, but I am not 100% sure this moment (human memory is immutable, right?)


I did not have the sync errors until I went to my desktop to reach the content there. Before then I had seen multiple instances the syncs were working. I posted as soon as the expected results did not match expectation (when the dataloss was observed first).

I understand it was mostly written as a rant, I hear that, and it kinda was. I'll gladly share more info, environmental or otherwise, but right now not sure which more info to add that is helpful. So yeah I'm game for helping you/others help me, no worries there. :)

I do appreciate your response and help on this though, thanks! :D

**Comment by joshtrichards** [comment]
To clarify, by "desktop" are you referring to accessing Notes via the Nextcloud Server Web UI? Or are you referring to something having to do with the Nextcloud Desktop sync client?

**Comment by BloodyIron** [comment]
> To clarify, by "desktop" are you referring to accessing Notes via the Nextcloud Server Web UI? Or are you referring to something having to do with the Nextcloud Desktop sync client?

No, Desktop means both webGUI and the desktop sync client. As in, another computer separate from the device the content was syncing _FROM_, not _specifically only one way of accessing nextCloud on another device_.

**Comment by jayjay73** [comment]
I can confirm a data loss / sync issue. I just installed notes together with the IOS app and within minutes of creating a test note and using the IOS app and the web app interchangeably I was seeing lots of version mismatches. 

After half an hour I gave up and de-installed everything notes related. I'd like to help you out, guys, but if this basic level of functionality is not working, I'm out. The "user base of approximately 20.000 users" that don't experience this are no help here. 

All the best.

**Comment by jasonblewis** [comment]
I'm experiencing the same issue with the iOS notes app. Fails to sync and overwrites newer version on the server with older version from the device.

**Comment by joshtrichards** [comment]
Does someone on this thread want to actually fill out the bug report template and/or provide some logs (maybe in your own dedicated Issue since the OP didn't even mention iOS) and also give basic context about their environment, etc.?

So far I'm just seeing vague "error" and "fails" messages. We can't fix what we can neither reproduce nor even guess at the cause of. We need info. :-)

**Comment by jasonblewis** [comment]
thanks for pointing out the error of my ways :smile: . I will endeavour to gather more data and fill in the bug report.

**Comment by himijendrix24** [comment]
I have the same issue on windows and android very randomly.

Example: Make some notes on the desktop, even hit STR+S to force upload.
Open nextcloud-notes on android, notes are not synced and old. 
Write something and the previously made notes are gone, which is really hard to notice.
Usually this only happens once and after this it works perfectly as expected, even without hitting STR+S.

Device A: windows + firefox + notes app version 4.11.0

Device B: Moto g84 Android 14 + notes app 4.3.1 + nextcloud app 3.31.0RC1 
All premissions garanted, unrestricted battery and data usage
background sync turned on

nextcloud is hosted on hetzner (storage-share nx11 version 29.0.9).

------------------------------------------

I guess the problem is on android and maybe the nextcloud-notes app should check for updates more proactive?
Maybe its also a problem with hetzner?

**Comment by cheywood** [comment]
Known issue with the Android app for eight years, see here for discussion https://github.com/nextcloud/notes-android/issues/161
